ContactList/ContactList/ViewControllerForUpdate.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  ViewControllerForUpdate.swift
+//  ContactList
+//
+//  Created by Maksym Humeniuk on 5/6/19.
+//  Copyright © 2019 Maksym Humeniuk. All rights reserved.
+//
+
+import UIKit
+
+class ViewControllerForUpdate: UIViewController {
+    
+    
+    @IBOutlet weak var firstNameTF: UITextField!
+    
+    @IBOutlet weak var lastNameTF: UITextField!
+    
+    @IBOutlet weak var phoneTF: UITextField!
+    
+    @IBOutlet weak var emailTF: UITextField!
+    
+    @IBOutlet weak var imageAreaTF: UIImageView!    
+    
+    @IBOutlet var forUpdate: UIView!
+    
+    @IBOutlet weak var saveButton: UIBarButtonItem!
+    
+    @IBOutlet weak var removeButton: UIBarButtonItem!
+    
+    
+    var currentPersonForEditing = Person()
+    var currentPersonCopy = Person()
+    
+    var handler: ContactListHandler?
+    var callback: ((Person, Bool) -> Void)?
+    
+    var changedArrayItemIndex: Int?
+    var fieldsCheckingIsOk: Bool!
+    
+    let picker = UIImagePickerController()
+    
+    @IBAction func cancelAction(_ sender: Any) {
+        dismiss(animated: true, completion: nil)
```

Comment:
You need to end editing before screen dismissal - keyboard remains on screen otherwise